crates/valence_protocol/src/inventory.rs

```diff
@@ -26,6 +26,7 @@ pub enum InventoryKind {
     Smoker,
     Cartography,
     Stonecutter,
+    Player,
```

Comment:
This enum is supposed to match the [window type in the open inventory packet](https://wiki.vg/Inventory). By adding another variant it wouldn't match the packet anymore (Notice the `Encode` and `Decode` derives).

What should happen if a client "opens" a player inventory? Should they be shown a generic 9x3? 9x4 for the hotbar? Nothing at all? What happens if a client has an `Inventory` component added to it that isn't a player inventory (by removing the default one maybe)?  Should `update_player_inventories` check for this and skip over it?

It seems like the player inventory should be an inherent part of `Client` or perhaps a separate component.

Reply:
I _really_ don't like the idea of `Client` having slots or a special inventory that's not an `Inventory`. For one, it would make it impossible to "opt-out" of behavior that depends on it, like `update_player_inventories`, if a user decides they don't need it. Also, it just opens us up to implementing the same inventory code twice.

Adding multiple components of the same type is currently [not supported by bevy_ecs, and is still being debated](https://github.com/bevyengine/bevy/issues/1527).  We shouldn't worry about that until it actually becomes an issue.

Perhaps we should have a `WindowType` enum that matches what's supposed to go in the packet, and the `InventoryKind` enum as it currently is. I also think that for now, it would be best to just print a warning and/or return a `Result::Err` if a client tries to open a `Player` inventory. We can always change it later to open a 9x3 or 9x4 if we decide we want to do that.

Reply:
> Perhaps we should have a WindowType enum that matches what's supposed to go in the packet, and the InventoryKind enum as it currently is. I also think that for now, it would be best to just print a warning and/or return a Result::Err if a client tries to open a Player inventory. We can always change it later to open a 9x3 or 9x4 if we decide we want to do that.

Alright that sounds like the way to go.